pkg/cmd/pulumi/plugin_install.go

```diff
@@ -103,6 +103,18 @@ func newPluginInstallCmd() *cobra.Command {
 					Checksums:         checksums,
 				}
 
+				// bundled plugins are not installable with this command.
```

Comment:
Annoyingly I'm not sure we should use "bundled" to decided this.
java, dotnet, and yaml are all bundled (that is we expect to see them side-by-side with the CLI binary) but they _are_ installable via `pulumi plugin install` and we probably are going to make use of that testing the install these things as we move to making them true plugins.
But maybe for decent UX just disable this check if PULUMI_DEV is set, I don't think any users will be depending on those installs yet but DEV will mean we can still quickly hook into it for testing.